File: projetoJava/src/FuncionarioHorista.java
class FuncionarioHorista extends Funcionario{
    private int horasTrabalhadas;
    private double valorHora;

    public FuncionarioHorista(String nome, int horasTrabalhadas, double valorHora) {
        this.nome = nome;
        this.horasTrabalhadas = horasTrabalhadas;
        this.valorHora = valorHora;


    }

    @Override
    public double calcularSalario() {
        return horasTrabalhadas * valorHora;
    }
}

File: projetoGo/funcionarioHorista.go
package main

type FuncionarioHorista struct {
	nome             string
	horasTrabalhadas float64
	valorPorHora     float64
}

func (f FuncionarioHorista) calcularSalario() float64 {
	return f.horasTrabalhadas * f.valorPorHora
}
